app/src/test/java/com/hover/stax/data/remote/AuthApiTest.kt

```diff
@@ -0,0 +1,319 @@
+                code = "76233958-77a5-43fc-9b3f-ca2d0d0ce54f",
+                action = "896fa22f-d273-475b-80ee-e7553d9f9a15"
+            ),
+            status = "152b1eff-f55a-4b57-9f45-0eaf5314d3c5"
+        )
+        val response = """{
+            "redirect_uri" : {
+                "code" : "76233958-77a5-43fc-9b3f-ca2d0d0ce54f",
+                "action" : "896fa22f-d273-475b-80ee-e7553d9f9a15"
+            },
+            "status" : "152b1eff-f55a-4b57-9f45-0eaf5314d3c5"
+        }""".trimIndent()
+
+        val mockEngine = MockEngine {
+            respond(
+                content = response,
+                status = HttpStatusCode.OK,
+                headersOf(HttpHeaders.ContentType, "application/json")
+            )
+        }
+
+        val api = StaxApi(
+            KtorClientFactory(
+                DefaultTokenProvider(testDataStore),
+                environmentProvider
+            ).create(mockEngine), environmentProvider
+        )
+
+        val actual = runBlocking { api.authorize(authRequest) }
+
+        assertThat(actual).isEqualTo(authResponse)
+    }
+
+    @Test
+    fun `test token request is successful when authorization code is correct`() {
+        val tokenRequest = fixture<TokenRequest>()
+        val tokenResponse = TokenResponse(
+            accessToken = "eyJpc3MiOiJTdGF4IE1vYmlsZSIsImlhdCI6MTY2MTc4NTkxMSwianRpIjoiZDFiYmEwZGItOWNiOC00OWUxLWEzNTItODk3NzYxMDhjYWZkIiwidXNlciI6IntcImRhdGFcIjp7X",
+            refreshToken = "A3_rBRCZUtL3i0h-y2_HWtw1icW9ZUaeH9Fq7R42GXg",
+            scope = "write",
+            createdAt = 1661785911,
+            tokenType = "Bearer",
+            expiresIn = 7200
+        )
+        val response = """{
+                "access_token" : "eyJpc3MiOiJTdGF4IE1vYmlsZSIsImlhdCI6MTY2MTc4NTkxMSwianRpIjoiZDFiYmEwZGItOWNiOC00OWUxLWEzNTItODk3NzYxMDhjYWZkIiwidXNlciI6IntcImRhdGFcIjp7X",
+                "refresh_token" : "A3_rBRCZUtL3i0h-y2_HWtw1icW9ZUaeH9Fq7R42GXg",
+                "scope" : "write",
+                "created_at" : 1661785911,
+                "token_type" : "Bearer",
+                "expires_in" :  7200
+        }""".trimIndent()
+
+        val mockEngine = MockEngine {
+            respond(
+                content = response,
+                status = HttpStatusCode.OK,
+                headersOf(HttpHeaders.ContentType, "application/json")
+            )
+        }
+
+        val api = StaxApi(
+            KtorClientFactory(
+                DefaultTokenProvider(testDataStore),
+                environmentProvider
+            ).create(mockEngine), environmentProvider
+        )
+
+        val actual = runBlocking { api.fetchToken(tokenRequest) }
+
+        assertThat(actual).isEqualTo(tokenResponse)
+    }
+
+    @Test
+    fun `test token revoke is successful when revoke token is correct`() {
+        val tokenRevoke = fixture<RevokeTokenRequest>()
+
+        val response = """{}""".trimIndent()
+
+        val mockEngine = MockEngine {
```

Comment:
Would it improve this test by adding the mock engine and API initialization to @Before test function?

Reply:
I was thinking of doing. Biggest reason we can't do that is
*  We need to create different scenarios of the mock engine, depending on the response we want - for example, we can configure an engine to return error 500, or 200 with a response depending on the test scenario.

This made it hard to init the mock engine and the api outside the context of the test itself